module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -1656,7 +1656,12 @@ zfs_readdir(struct inode *ip, zpl_dir_context_t *ctx, cred_t *cr)
  */
 /* ARGSUSED */
 int
+#ifdef HAVE_GENERIC_FILLATTR_USERNS
+zfs_getattr_fast(struct user_namespace *user_ns, struct inode *ip,
+    struct kstat *sp)
+#else
 zfs_getattr_fast(struct inode *ip, struct kstat *sp)
+#endif
```

Comment:
Since this is an internal interface under our control I think it'd be cleaner to always pass the user_ns.  One way to do this would be to emulate the kernel's behavior and always pass the initial credential when an alternate user_ns wasn't provided `kcred->user_ns`.

Reply:
Sounds good, thanks for the tip - I'll work up a commit that addresses these later tonight and push it to the branch.

Reply:
Let me know what you thing about the solution in 04d4c20

Reply:
Yeah, I think that'll work nicely.

Reply:
Err 58e2ab2, because I missed the change in the header.